daemon/dpudaemon.go

```diff
@@ -106,11 +112,34 @@ func (d *DpuDaemon) Listen() (net.Listener, error) {
 }
 
 func (d *DpuDaemon) ListenAndServe() error {
-	lis, err := d.Listen()
+	done := make(chan error, 1)
+	listener, err := d.Listen()
+
 	if err != nil {
+		d.log.Error(err, "Failed to listen")
 		return err
 	}
-	return d.Serve(lis)
+	go func() {
```

Comment:
This implementation does not work:
```package main

import (
	"errors"
	"fmt"
	"time"
)

func something(c chan error) {
	fmt.Println("HELLO1")
	c <- nil
}

func something3(c chan error) {
	fmt.Println("HELLO1")
	time.Sleep(5 * time.Second)
	c <- errors.New("an error")
}

func something2() error {
	c := make(chan error, 1)
	go something(c)
	go something3(c)
	return <-c
}
func main() {
	err := something2()
	fmt.Println(err)
}
```
It does not wait for all the go routines to finish, you need a wait group.

Reply:
Something like this works:
```
package main

import (
	"errors"
	"fmt"
	"sync"
	"time"
)

func test1(wg *sync.WaitGroup, c chan error) {
	defer wg.Done()
	fmt.Println("HELLO1")
	c <- nil
}

func test2(wg *sync.WaitGroup, c chan error) {
	defer wg.Done()
	fmt.Println("HELLO1")
	time.Sleep(5 * time.Second)
	c <- errors.New("an error")
}

func testing() chan error {
	c := make(chan error, 2)
	var wga sync.WaitGroup
	wga.Add(1)
	go test1(&wga, c)
	wga.Add(1)
	go test2(&wga, c)
	wga.Wait()
	close(c)
	return c
}
func main() {
	errors := testing()
	for err := range errors {
		fmt.Println(err)
	}
}
```